File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/agenda/AgendaDayPagerAdapter.java
package com.mentalmachines.droidcon_boston.views.agenda;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.app.FragmentManager;

import com.mentalmachines.droidcon_boston.data.ScheduleDatabase;

public class AgendaDayPagerAdapter extends FragmentPagerAdapter {
    @SuppressWarnings("FieldCanBeLocal")
    private final int PAGE_COUNT = 2;
    private String tabTitles[] = new String[] { "Day 1", "Day 2" };

    public AgendaDayPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Override
    public Fragment getItem(int position) {
        return AgendaDayFragment.newInstance(
                (position == 0) ? ScheduleDatabase.MONDAY : ScheduleDatabase.TUESDAY
        );
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return tabTitles[position];
    }
}

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/agenda/AgendaDayPagerAdapter.kt
package com.mentalmachines.droidcon_boston.views.agenda

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.mentalmachines.droidcon_boston.BuildConfig
import com.mentalmachines.droidcon_boston.data.Schedule

class AgendaDayPagerAdapter internal constructor(
    fm: FragmentManager,
    private val myAgenda: Boolean
) :
    FixedFragmentStatePagerAdapter(fm) {

    private val pageCount = 2
    private val tabTitles = arrayOf("Day 1", "Day 2")

    override fun getCount(): Int {
        return pageCount
    }

    override fun getItem(position: Int): Fragment {
        val dayString = if (position == 0) {
            BuildConfig.EVENT_DAY_ONE_STRING
        } else {
            BuildConfig.EVENT_DAY_TWO_STRING
        }

        return AgendaDayFragment.newInstance(
            myAgenda,
            dayString
        )
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return tabTitles[position]
    }

    override fun updateFragmentItem(position: Int, fragment: Fragment) {
        if (fragment is AgendaDayFragment) {
            fragment.updateList()
        }
    }

    override fun getFragmentItem(position: Int): Fragment {
        return getItem(position)
    }
}
